### PR TITLE
You can fish out of hydroponics trays

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Adds a fishing spot to hydroponics trays. 

Chances 

- 33% nothing
- 33% grass
- 20% random seed (discludes rarer seeds, such as gatfruit and most mutations) 
- 10% grass seeds
- 1% strange seed
- 1% axolotl
- 1% frog

Lightly inspired by something (I think) is possible on Goon, but obviously entirely new code. 

## Why It's Good For The Game

Mostly just for laughs, they're basins of water so surely you can get something out of it. 

## Changelog

:cl: Melbert
add: Fishers can now try their luck at fishing out of hydroponics basins. 
/:cl:

### CODE REVIEW

**Comment by the-orange-cow** [comment]
Koibeans missing out, smh.

File: code/modules/fishing/sources/source_types.dm [hunk 1]
```diff
@@ -310,3 +310,61 @@
...
+		/obj/item/seeds/grass = 4,
+		RANDOM_SEED = 4,
+		/obj/item/seeds/random = 1,
+		/mob/living/basic/frog = 1,
+		/mob/living/basic/axolotl = 1,
+	)
+	fishing_difficulty = FISHING_DEFAULT_DIFFICULTY - 10
+
+/datum/fish_source/hydro_tray/reason_we_cant_fish(obj/item/fishing_rod/rod, mob/fisherman, atom/parent)
+	if(istype(parent, /obj/machinery/hydroponics/constructable))
```
Comment (san7890) [inline code/modules/fishing/sources/source_types.dm:338]: early `return ..()` here?

**Review by san7890 — CHANGES_REQUESTED** [review]
nit

File: code/modules/fishing/sources/source_types.dm [hunk 2]
```diff
@@ -310,3 +310,61 @@
...
+			return "There's a plant growing in [parent]."
+
+	return ..()
+
+/datum/fish_source/hydro_tray/spawn_reward(reward_path, mob/fisherman, turf/fishing_spot)
+	if(reward_path != RANDOM_SEED)
+		var/atom/movable/created_reward = ..()
+		if(ismob(created_reward))
+			created_reward.name = "small [created_reward.name]"
+			created_reward.transform = created_reward.transform.Scale(0.75)
```
Comment (Ghommie) [inline code/modules/fishing/sources/source_types.dm:352]:
```suggestion
			var/mob/living/mob_reward = created_reward
			mob_reward.name = "small [created_reward.name]"
			mob_reward.update_transform(0.75)
```

**Review by Ghommie — CHANGES_REQUESTED** [review]
Looks good to me, except for the resizing.

**Pull request merged**